src/viking.js

```diff
@@ -1,11 +1,168 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(theFirstArg, theSecondArg) {
+    this.health = theFirstArg;
+    this.strength = theSecondArg;
+  }
+  attack() {
+    return this.strength;
+  }
+
+  receiveDamage(amountOfDamage) {
+    this.health -= amountOfDamage;
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(theFirstArg, theSecondArg, theThirdArg) {
+    super(theSecondArg, theThirdArg);
+    this.name = theFirstArg;
+  }
+
+  receiveDamage(amount) {
+    super.receiveDamage(amount);
+    if (this.health > 0) {
+      return `${this.name} has received ${amount} points of damage`;
+    } else {
+      return `${this.name} has died in act of combat`;
```

Comment:
nesses casos vc nem precisa usar "else" se quiser, depois do if se quiser mandar direto o "return `${this.name} has died in act of combat`", também da certo :wink: